API/NoDocs/Microsoft.AspNetCore.Telemetry/LoggingOptions.cs

```diff
@@ -0,0 +1,39 @@
+// Assembly 'Microsoft.AspNetCore.Telemetry.Middleware'
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.Compliance.Classification;
+using Microsoft.Extensions.Http.Telemetry;
+using Microsoft.Shared.Data.Validation;
+
+namespace Microsoft.AspNetCore.Telemetry;
+
+public class LoggingOptions
+{
+    public bool LogRequestStart { get; set; }
+    public bool LogBody { get; set; }
+    public IncomingPathLoggingMode RequestPathLoggingMode { get; set; }
+    [Experimental("EXTEXP0003", UrlFormat = "https://aka.ms/dotnet-extensions-warnings/{0}")]
+    public HttpRouteParameterRedactionMode RequestPathParameterRedactionMode { get; set; }
+    [TimeSpan(1, 60000)]
+    public TimeSpan RequestBodyReadTimeout { get; set; }
+    [Range(1, 1572864)]
+    public int BodySizeLimit { get; set; }
+    [Required]
+    public IDictionary<string, DataClassification> RouteParameterDataClasses { get; set; }
+    [Required]
+    public IDictionary<string, DataClassification> RequestHeadersDataClasses { get; set; }
+    [Required]
+    public ISet<string> RequestBodyContentTypes { get; set; }
+    [Required]
+    public IDictionary<string, DataClassification> ResponseHeadersDataClasses { get; set; }
+    [Required]
+    public ISet<string> ResponseBodyContentTypes { get; set; }
+    [Experimental("EXTEXP0003", UrlFormat = "https://aka.ms/dotnet-extensions-warnings/{0}")]
+    [Required]
+    public ISet<string> ExcludePathStartsWith { get; set; }
```

Comment:
There are a lot of `ISet<T>` in other option classes.

Reply:
This is not about performance, this is about indicating the fact all values should be unique and unordered. IList doesn't have those semantics.

Reply:
I don't think developers know or care about the exact semantics of how the values are used. I bring it up because it's my hunch that they are more familiar and comfortable with configuring an `IList<T>` than an `ISet<T>` on an options type.

But I don't feel strongly about this.

Reply:
Leaving this as ISet.